feat: extend benchmark endpoint by trend50d and trend200d

Dhoni77 opened the pull request:
Closes #2567

- Calculate benchmark trends for trend50d and trend200d
- Added interface and type

Dhoni77 commented:
Hi @dtslvr could you please check if the expected logic is correct in the above pr.
If so I would further refactor the code.
Thanks

dtslvr commented on apps/api/src/app/benchmark/benchmark.service.ts:
```diff
@@ -45,6 +49,31 @@ export class BenchmarkService {
     return 0;
   }
 
+  public async getBenchMarkTrends(dataSource: DataSource, symbol: string) {
```
Please use `benchmark` instead of `benchMark` 😛

dtslvr commented on apps/api/src/app/benchmark/benchmark.service.ts:
```diff
@@ -45,6 +49,31 @@ export class BenchmarkService {
   }
 
+  public async getBenchMarkTrends(dataSource: DataSource, symbol: string) {
+    return this.marketDataService
```
`const historicalData = await this.marketDataService...` instead of chaining

dtslvr commented on apps/api/src/app/benchmark/benchmark.service.ts:
```diff
@@ -45,6 +49,31 @@ export class BenchmarkService {
+        }
+      })
+      .then((historicalData) => {
+        const fiftyDayAvg = calculateBenchmarkTrend(historicalData, 50);
```
`Average` instead of `Avg`

dtslvr commented on apps/api/src/app/benchmark/benchmark.service.ts:
```diff
@@ -93,15 +130,18 @@ export class BenchmarkService {
           allTimeHigh: {
-            date: allTimeHigh.date,
-            performancePercent: performancePercentFromAllTimeHigh
+            date: allTimeHigh?.date || new Date(),
```
`??` instead of `||`

dtslvr commented on libs/common/src/lib/types/benchmark-trend-type.type.ts:
Can you add it to `libs/common/src/lib/types/index.ts`?

dtslvr reviewed (changes requested):
Thanks for providing this draft, @Dhoni77.

I have added `^GSPC` as _Market Data_ in the _Admin Control_ panel, gathered historical data and checked the results against these [moving averages](https://www.barchart.com/stocks/quotes/$SPX/technical-analysis) (red for `DOWN`, green for `UP`). The values match 😎

In the first request, one value of `50d` and `200d` was `null`, after reloading the endpoint, both values were there. Have you seen this behaviour before?

I have already added some comments for the refactoring. If you are interested, you are invited to write a test for the moving average function in a follow-up task.

Thank you once again for your dedication!

Dhoni77 commented:
> Thanks for providing this draft, @Dhoni77.
> 
> I have added `^GSPC` as _Market Data_ in the _Admin Control_ panel, gathered historical data and checked the results against these [moving averages](https://www.barchart.com/stocks/quotes/$SPX/technical-analysis) (red for `DOWN`, green for `UP`). The values match 😎
> 
> In the first request, one value of `50d` and `200d` was `null`, after reloading the endpoint, both values were there. Have you seen this behaviour before?
> 
> I have already added some comments for the refactoring. If you are interested, you are invited to write a test for the moving average function in a follow-up task.
> 
> Thank you once again for your dedication!

Hi @dtslvr, If we need to calcuate the 50day trend we need the data for last 100 days and for 200d trend we need data for 400 days. So if data is not present I am returning null. Is this correct ?

dtslvr commented:
> Is this correct ?

Yes, that makes sense. I think data was already there, but still got `null`.
